File: Prefix Sum/3355. Zero Array Transformation I.cpp
class Solution {
public:
    bool isZeroArray(vector<int>& nums, vector<vector<int>>& queries) 
    {
        int n = nums.size();
        vector<int> difArray(n + 1);
        for(auto &que : queries)
        {
            difArray[que[0]] += 1;
            difArray[que[1] + 1] -= 1;
        }

        for(int i = 1; i < n; i++)
            difArray[i] += difArray[i - 1];
        
        for(int i = 0; i < n; i++)
            if(nums[i] > difArray[i]) return false;
            
        return true;    
    }
};

File: Prefix Sum/3355. Zero Array Transformation I.py
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        difArray = [0] * (n + 1)

        for l, r in queries:
            difArray[l] += 1
            difArray[r + 1] -= 1

        for i in range(1, n):
            difArray[i] += difArray[i - 1]
        
        for i in range(n):
            if nums[i] > difArray[i]:
                return False
        return True